src/Core/Models/Data/EnvironmentUrlData.cs

```diff
@@ -44,5 +46,63 @@ public class EnvironmentUrlData
                 && string.IsNullOrEmpty(Notifications)
                 && string.IsNullOrEmpty(WebVault)
                 && string.IsNullOrEmpty(Events);
+
+        public Region Region
+        {
+            get
+            {
+                if (Base == Region.US.BaseUrl())
+                {
+                    return Region.US;
+                }
+                if (Base == Region.EU.BaseUrl())
+                {
+                    return Region.EU;
+                }
+                return Region.SelfHosted;
+            }
+        }
+
+        public EnvironmentUrlData FormatUrls()
+        {
+            return new EnvironmentUrlData
+            {
+                Base = FormatUrl(Base),
+                Api = FormatUrl(Api),
+                Identity = FormatUrl(Identity),
+                Icons = FormatUrl(Icons),
+                Notifications = FormatUrl(Notifications),
+                WebVault = FormatUrl(WebVault),
+                Events = FormatUrl(Events)
+            };
+        }
+
+        private string FormatUrl(string url)
+        {
+            if (string.IsNullOrWhiteSpace(url))
+            {
+                return null;
+            }
+            url = Regex.Replace(url, "\\/+$", string.Empty);
+            if (!url.StartsWith("http://") && !url.StartsWith("https://"))
+            {
+                url = string.Concat("https://", url);
+            }
+            return url.Trim();
+        }
+
+        public string ParseEndpoint()
+        {
+            var url = WebVault ?? Base ?? Api ?? Identity;
+            if (!string.IsNullOrWhiteSpace(url))
+            {
+                if (url.Contains(Region.US.Domain()) || url.Contains(Region.EU.Domain()))
+                {
+                    return CoreHelpers.GetDomain(url);
+                }
+                return CoreHelpers.GetHostname(url);
+            }
+            return string.Empty;
+        }
```

Comment:
🎨 Could we change the name? As a consumer, I wouldn't know what to expect when calling `ParseEndpoint()`, I'd need to enter and see what it does to know what it will return.
Also, I feel like changing the order would be beneficial for reading:
```suggestion
        public string ParseEndpoint()
        {
            var url = WebVault ?? Base ?? Api ?? Identity;
            if (string.IsNullOrWhiteSpace(url))
            {
                return string.Empty;
            }
            if (url.Contains(Region.US.Domain()) || url.Contains(Region.EU.Domain()))
            {
                return CoreHelpers.GetDomain(url);
            }
            return CoreHelpers.GetHostname(url);
        }
```

Reply:
🎨 @andrebispo5 Could the name of `ParseEndpoint()` be changed? for the reasons explained above.